Home_Work_20/Fight.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Fight.h"
+
+#include "Item.h"
+#include "Modifier.h"
+#include "Munchkin.h"
+#include "Monster.h"
+#include "Runaway.h"
+#include "Victory.h"
+#include <cassert>
+
+void Fight::start()
+{
+	m_result = FightResult::InProgress;
+
+	calculateMunchkinPower();
+	calculateMonsterPower();
+}
+
+void Fight::applyModifier(int choice)
+{
+	Modifier* modifier = m_munchkin->popModifier(choice);
+	if (modifier == nullptr)
+	{
+		//DONE
+		assert(modifier != nullptr && "Modifier in deck is nullptr");
+		//DONE
+		return;
+	}
+
+	modifier->apply(m_munchkin, m_monster);
+
+	m_MunchkinPower += modifier->getMunchkinModifier();
+	m_monsterPower += modifier->getMonsterModifier();
+	m_monsterIsFriend = modifier->isMonsterFriend();
+}
+
+void Fight::runawayFlow()
+{
+	Runaway* policy = m_monster->getRunawayPolicy();
+	policy->apply(m_munchkin);
+
+	m_result = FightResult::MonsterWon;
+}
+
+void Fight::victoryFlow(ItemDeck* items, ModifierDeck* modifiers)
+{
+	
+	m_munchkin->updateLevelBy(1);
```

Comment:
It should be part of the victory policy. For example, as a default VictoryPolicy if no specific was provided for the Monster